internal/api/databases.go

```diff
@@ -53,3 +56,40 @@ func handleGetDatabases(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	outputJSON(c, w, databases)
 }
+
+// handleUpdateDatabase responds to PUT /api/database/{database}, updating the
+// database configuration values.
+func handleUpdateDatabase(c *Context, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	databaseID := vars["database"]
+	c.Logger = c.Logger.WithField("database", databaseID)
+
+	database, status, unlockOnce := lockDatabase(c, databaseID)
+	if status != 0 {
+		w.WriteHeader(status)
+		return
+	}
+	defer unlockOnce()
+
+	patchDatabaseRequest, err := model.NewPatchDatabaseRequestFromReader(r.Body)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to decode request")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
```

Comment:
nit: Maybe we could switch those two around to eliminate unnecessary locking if the request body is invalid. I think we do it the other way around in different places too.